### PR TITLE
Consolidate dropdown state

### PR DESCRIPTION
**Pull request by marthacryan**
* Consolidates the `UiState` fields `currOpenDropdown` and `toolbarDropdown` to be one field. 
* Updates the type to `Dropdown`, which is either a string to id the dropdown (for toolbar dropdowns) or a `row` and `col` to identify the row, column, or cell that was right-clicked on. 
* The purpose of this PR is to prevent having multiple dropdowns open at once

### CODE REVIEW

**Review by naterush — COMMENTED** [review]
This works excellent, and is great polish. 

Screen Shot 2023-11-30 at 4 52 51 PM

For some reason, it doesn't work on the format dropdown. Any idea why?

File: mitosheet/src/mito/components/elements/Select.tsx [hunk 1]
```diff
@@ -159,7 +165,10 @@ const Select = (props: SelectProps): JSX.Element => {
             </div>
             <Dropdown
                 display={displayDropdown}
-                closeDropdown={() => setDisplayDropdown(false)}
+                closeDropdown={() => {
+                    props.onChangeDisplay?.(false)
+                    setDisplayDropdown(false)
```
[the remark this answers is not in the document]
Comment (naterush) [reply mitosheet/src/mito/components/elements/Select.tsx:170]: Part of me wonders if an effect on `display` would do this -- but perhaps it would loop?

**Comment by naterush** [comment]
Screen Shot 2023-12-04 at 10 09 20 AM

Not sure if you addressed this in a different PR, but a reminder that the spacing changes on the main text in dropdown items. I'd prefer to move it back to where it was - it feels a bit too left to me currently.

If you're tracking in another PR, ignore.

**Comment by naterush** [comment]
Closing logic working very well.

File: mitosheet/src/mito/components/endo/ColumnHeaderDropdown.tsx [hunk 2]
```diff
@@ -17,31 +17,43 @@ import { getPropsForContextMenuDropdownItem } from './utils';
...
     return (
         <Dropdown
-            display={props.display}
-            closeDropdown={() => props.setOpenColumnHeaderDropdown(false)}
+            display={display}
+            closeDropdown={() => {
+                props.setUIState((prevUIState) => {
+                    // If the dropdown is open, then close it. Otherwise, don't change the state. 
+                    const openColumnHeaderDropdown = typeof prevUIState.currOpenDropdown == 'object' ? prevUIState.currOpenDropdown.column : undefined;
+                    const display = openColumnHeaderDropdown === props.column;
```
[the remark this answers is not in the document]
Comment (marthacryan) [reply mitosheet/src/mito/components/endo/ColumnHeaderDropdown.tsx:50]: True!

File: mitosheet/src/mito/components/elements/Select.tsx [hunk 3]
```diff
@@ -142,11 +147,12 @@ const Select = (props: SelectProps): JSX.Element => {
                     /* 
                         Only change the visibility of the dropdown if the button is 
                         not already open because the dropdown handles closing itself
```
[the remark this answers is not in the document]
Comment (marthacryan) [reply mitosheet/src/mito/components/elements/Select.tsx:149]: Yeah I think it actually does still make sense - I think it's saying that when the dropdown buttons are pressed, they automatically close the dropdown, so there's no need for updating state. This actually still applies!

**Pull request merged**